app/models/api_docs/service.rb

```diff
@@ -20,6 +20,8 @@ class ApiDocs::Service < ApplicationRecord
   validates :body, length: { maximum: 4294967295, allow_blank: true }
   validate :service_belongs_to_account, if: -> { service_id.present? && service_id_changed? }
 
+  before_validation(on: :create) { self.account ||= service&.account }
```

Comment:
I like this a lot! 💪

Reply:
in which case do we need `&.` operator?

Reply:
ApiDocs::Service.new(... without account and without service. Which will be invalid, but it shouldn't be for "NoMethodError for nil" 😂

Reply:
Same cases we need the `if` in 
https://github.com/3scale/porta/blob/7da01b64214109778b8030d770bde282f6c3aca3/app/models/api_docs/service.rb#L21

Reply:
Ah right I totally forgot that service is not required.

But account is required
https://github.com/3scale/porta/blob/7da01b64214109778b8030d770bde282f6c3aca3/app/models/api_docs/service.rb#L8

Sop with this you can create an ApiDoc like this:

```ruby
ApiDocs::Service.create(service: service, ...)
```

whereas before we enforce

```ruby
ApiDocs::Service.create(service: service, account: account...)
```

IMHO enforcing the API is better: We need an account.

Reply:
I understand it is for this: https://github.com/3scale/porta/pull/119/files#diff-f55ef40b403ec9f0c09f95cd2825bcf5R36 not to repeat the `account` attribute

Reply:
Correct. Better `service.api_docs_service.build(...)` than `service.api_docs_service.build(account: service.account, ...)`